pkgs/desktops/cinnamon/xapps/default.nix

```diff
@@ -0,0 +1,50 @@
+{ fetchFromGitHub, glib, gobject-introspection, gtk3, libgnomekbd, libxklavier, meson, ninja, pkgconfig, python3, python3Packages, stdenv, vala, wrapGAppsHook }:
+
+stdenv.mkDerivation rec {
+  pname = "xapps";
+  version = "1.6.4";
+  enableParallelBuilding = true;
+
+  src = fetchFromGitHub {
+    owner = "linuxmint";
+    repo = pname;
+    rev = version;
+    sha256 = "08c8j7wv56zf8mfgjfhd0wc1bbrw0dj1lmlawvsxwfajnnyjsm0d";
+  };
+
+  NIX_CFLAGS_COMPILE = [ "-I${glib.dev}/include/gio-unix-2.0" ]; # TODO: https://github.com/NixOS/nixpkgs/issues/36468
+
+  patches = [
+    ./py-override.patch # details see https://github.com/linuxmint/xapps/pull/65
+  ];
```

Comment:
Can you fetch the patches from my PR using `fetchpatch`.
This should also fix the issue where you have to substitute `prefix` below so it doesn't install into `/`.

Reply:
I also noticed this expression produces a messed up directory tree
```
└── usr
    ├── bin
    │   ├── pastebin
    │   ├── upload-system-info
    │   └── xfce4-set-wallpaper
    └── share
        └── icons
            └── hicolor
                └── scalable
                    ├── actions
                    │   ├── add-files-to-archive-symbolic.svg
                    │   ├── category-search-symbolic.svg
                    │   ├── extract-archive-symbolic.svg
                    │   ├── media-mount-symbolic.svg
                    │   ├── view-bottom-pane-symbolic.svg
                    │   ├── view-left-pane-symbolic-rtl.svg
                    │   ├── view-left-pane-symbolic.svg
                    │   ├── view-right-pane-symbolic-rtl.svg
                    │   ├── view-right-pane-symbolic.svg
                    │   ├── view-top-pane-symbolic.svg
                    │   ├── xapp-go-history-next-symbolic-rtl.svg
                    │   ├── xapp-go-history-next-symbolic.svg
                    │   ├── xapp-go-history-previous-symbolic-rtl.svg
                    │   ├── xapp-go-history-previous-symbolic.svg
                    │   ├── xapp-search-entire-word-symbolic.svg
                    │   ├── xapp-search-wrap-symbolic.svg
                    │   ├── xapp-text-case-symbolic.svg
                    │   └── xapp-use-regex-symbolic.svg
                    └── categories
                        ├── xapp-prefs-behavior-symbolic.svg
                        ├── xapp-prefs-display-symbolic.svg
                        ├── xapp-prefs-plugins-symbolic.svg
                        ├── xapp-prefs-preview-symbolic.svg
                        └── xapp-prefs-toolbar-symbolic.svg
```
I believe my patch fixes that.

Reply:
When I apply your patch directly pulled from GH it causes a build error

```
evaluating file '/home/maciej/Projekte/nixpkgsv/pkgs/development/compilers/rust/default.nix'
evaluating file '/home/maciej/Projekte/nixpkgsv/pkgs/development/compilers/rust/rustc.nix'
evaluating file '/home/maciej/Projekte/nixpkgsv/pkgs/development/compilers/rust/bootstrap.nix'
evaluating file '/home/maciej/Projekte/nixpkgsv/pkgs/development/compilers/rust/binary.nix'
evaluating file '/home/maciej/Projekte/nixpkgsv/pkgs/tools/typesetting/docbook2x/default.nix'
evaluating file '/home/maciej/Projekte/nixpkgsv/pkgs/tools/text/sgml/opensp/default.nix'
evaluating file '/home/maciej/Projekte/nixpkgsv/pkgs/data/sgml+xml/schemas/xml-dtd/docbook/4.1.2.nix'
evaluating file '/home/maciej/Projekte/nixpkgsv/lib/deprecated.nix'
evaluating file '/home/maciej/Projekte/nixpkgsv/pkgs/development/compilers/rust/cargo.nix'
evaluating file '/home/maciej/Projekte/nixpkgsv/pkgs/build-support/rust/default.nix'
evaluating file '/home/maciej/Projekte/nixpkgsv/pkgs/development/libraries/git2/default.nix'
evaluating file '/home/maciej/Projekte/nixpkgsv/pkgs/development/libraries/http-parser/default.nix'
these derivations will be built:
  /nix/store/y38bfm0a3wqvqqj7az49lmy3q6f1981p-xapps-1.6.4.drv
building '/nix/store/y38bfm0a3wqvqqj7az49lmy3q6f1981p-xapps-1.6.4.drv'...
unpacking sources
unpacking source archive /nix/store/qmkrvhx4rj4xn62865ai1m6bjf71lw09-source
source root is source
patching sources
applying patch /nix/store/hf5bxq0rayva772v03ayasspvgcxm3j7-65.diff
patching file files/meson.build
patching file icons/hicolor/scalable/actions/add-files-to-archive-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/actions/add-files-to-archive-symbolic.svg)
patching file icons/hicolor/scalable/actions/category-search-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/actions/category-search-symbolic.svg)
patching file icons/hicolor/scalable/actions/extract-archive-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/actions/extract-archive-symbolic.svg)
patching file icons/hicolor/scalable/actions/media-mount-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/actions/media-mount-symbolic.svg)
patching file icons/hicolor/scalable/actions/view-bottom-pane-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/actions/view-bottom-pane-symbolic.svg)
patching file icons/hicolor/scalable/actions/view-left-pane-symbolic-rtl.svg (renamed from files/usr/share/icons/hicolor/scalable/actions/view-left-pane-symbolic-rtl.svg)
patching file icons/hicolor/scalable/actions/view-left-pane-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/actions/view-left-pane-symbolic.svg)
patching file icons/hicolor/scalable/actions/view-right-pane-symbolic-rtl.svg (renamed from files/usr/share/icons/hicolor/scalable/actions/view-right-pane-symbolic-rtl.svg)
patching file icons/hicolor/scalable/actions/view-right-pane-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/actions/view-right-pane-symbolic.svg)
patching file icons/hicolor/scalable/actions/view-top-pane-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/actions/view-top-pane-symbolic.svg)
patching file icons/hicolor/scalable/actions/xapp-go-history-next-symbolic-rtl.svg (renamed from files/usr/share/icons/hicolor/scalable/actions/xapp-go-history-next-symbolic-rtl.svg)
patching file icons/hicolor/scalable/actions/xapp-go-history-next-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/actions/xapp-go-history-next-symbolic.svg)
patching file icons/hicolor/scalable/actions/xapp-go-history-previous-symbolic-rtl.svg (renamed from files/usr/share/icons/hicolor/scalable/actions/xapp-go-history-previous-symbolic-rtl.svg)
patching file icons/hicolor/scalable/actions/xapp-go-history-previous-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/actions/xapp-go-history-previous-symbolic.svg)
patching file icons/hicolor/scalable/actions/xapp-search-entire-word-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/actions/xapp-search-entire-word-symbolic.svg)
patching file icons/hicolor/scalable/actions/xapp-search-wrap-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/actions/xapp-search-wrap-symbolic.svg)
patching file icons/hicolor/scalable/actions/xapp-text-case-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/actions/xapp-text-case-symbolic.svg)
patching file icons/hicolor/scalable/actions/xapp-use-regex-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/actions/xapp-use-regex-symbolic.svg)
patching file icons/hicolor/scalable/categories/xapp-prefs-behavior-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/categories/xapp-prefs-behavior-symbolic.svg)
patching file icons/hicolor/scalable/categories/xapp-prefs-display-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/categories/xapp-prefs-display-symbolic.svg)
patching file icons/hicolor/scalable/categories/xapp-prefs-plugins-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/categories/xapp-prefs-plugins-symbolic.svg)
patching file icons/hicolor/scalable/categories/xapp-prefs-preview-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/categories/xapp-prefs-preview-symbolic.svg)
patching file icons/hicolor/scalable/categories/xapp-prefs-toolbar-symbolic.svg (renamed from files/usr/share/icons/hicolor/scalable/categories/xapp-prefs-toolbar-symbolic.svg)
patching file icons/meson.build
patching file meson.build
patching file meson_options.txt
patching file pygobject/meson.build
patching file scripts/meson.build
patching file scripts/pastebin (renamed from files/usr/bin/pastebin)
patching file scripts/upload-system-info (renamed from files/usr/bin/upload-system-info)
patching file scripts/xfce4-set-wallpaper (renamed from files/usr/bin/xfce4-set-wallpaper)
patching script interpreter paths in schemas/meson_install_schemas.py
schemas/meson_install_schemas.py: interpreter directive changed from "/usr/bin/python3" to "/nix/store/h3fn2vr6c5jh0mjf02ir26cxb6vcdxd8-python3-3.7.5-env/bin/python3"
substitute(): ERROR: file 'files/meson.build' does not exist
builder for '/nix/store/y38bfm0a3wqvqqj7az49lmy3q6f1981p-xapps-1.6.4.drv' failed with exit code 1
error: build of '/nix/store/y38bfm0a3wqvqqj7az49lmy3q6f1981p-xapps-1.6.4.drv' failed
```

Reply:
ok, nvm, seems I forgot to remove the prefix substitution